From 5e7f3336f7a924a0401934d66512c4073aa785fc Mon Sep 17 00:00:00 2001
From: Fatih Uzunoglu <fuzun54@outlook.com>
Date: Thu, 18 Jan 2024 17:30:26 +0200
Subject: [PATCH 4/7] Expose QRhiImplementation in QRhi

Exposing QRhiImplementation is necessary, because that
is how the IDCompositionDevice instance created by Qt
is accessed. This is only relevant for the Direct
Composition integration.
---
 src/gui/rhi/qrhi.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/gui/rhi/qrhi.h b/src/gui/rhi/qrhi.h
index f7a49dc292..a4c74dd837 100644
--- a/src/gui/rhi/qrhi.h
+++ b/src/gui/rhi/qrhi.h
@@ -1987,6 +1987,9 @@ protected:
 private:
     Q_DISABLE_COPY(QRhi)
     QRhiImplementation *d = nullptr;
+
+public:
+    QRhiImplementation* implementation() { return d; };
 };
 
 Q_DECLARE_OPERATORS_FOR_FLAGS(QRhi::Flags)
-- 
2.44.0

